astronomy/orbit_ground_track_body.hpp

```diff
@@ -0,0 +1,164 @@
+﻿#pragma once
+
+#include "astronomy/orbit_ground_track.hpp"
+
+#include <limits>
+
+#include "physics/apsides.hpp"
+
+namespace principia {
+namespace astronomy {
+namespace internal_orbit_ground_track {
+
+using geometry::Position;
+using geometry::Vector;
+using physics::ComputeNodes;
+using quantities::Mod;
+using quantities::UnwindFrom;
+using quantities::si::Radian;
+
+// Note that the origin of this celestial longitude is arbitrary: it is not the
+// node of the orbit around the sun (the equinox).  If |PrimaryCentred| is the
+// GCRS, this is the right ascension (with respect to the mean equinox of
+// J2000).
+template<typename PrimaryCentred>
+Angle CelestialLongitude(Position<PrimaryCentred> const& q) {
+  // TODO(egg): |ToSpherical| is wasteful, as we discard the latitude.
+  return (q - PrimaryCentred::origin).coordinates().ToSpherical().longitude;
+}
+
+// The resulting angle is not normalized.
+template<typename Iterator, typename Inertial>
```

Comment:
Why is this taking an `Iterator` (which could be anything) and not a frame (where the iterator would be `DiscreteTrajectory<Frame>::Iterator`)?

Reply:
Deduction.